### PR TITLE
nagios check improvement: support to enforce check for a whole cluster

### PR DESCRIPTION
**Pull request by dennyzhang**

### CODE REVIEW

File: bash/enforce_all_nagios_check/enforce_all_nagios_check.sh [hunk 1]
```diff
@@ -1,77 +1,155 @@
 #!/bin/bash
 ##-------------------------------------------------------------------
 ## File : enforce_all_nagios_check.sh
-## Author : Denny <denny@dennyzhang.com>
+## Author : Denny <denny.zhang001@gmail.com>
```
Comment (dennyzhang) [inline bash/enforce_all_nagios_check/enforce_all_nagios_check.sh]: @syrett Please change to author to your email.

File: bash/enforce_all_nagios_check/enforce_all_nagios_check.sh [hunk 2]
```diff
@@ -1,77 +1,155 @@
...
 ## Description :
 ## --
 ## Created : <2015-06-24>
-## Updated: Time-stamp: <2016-04-09 09:35:43>
+## Updated: Time-stamp: <2016-04-18 15:58:19>
 ##-------------------------------------------------------------------
```
Comment (dennyzhang) [inline bash/enforce_all_nagios_check/enforce_all_nagios_check.sh:11]: It's better we add sample usage as comments here

File: bash/enforce_all_nagios_check/enforce_all_nagios_check.sh [hunk 3]
```diff
@@ -1,77 +1,155 @@
...
-                    echo "skip check: $command"
-                    skipped_checks="${skipped_checks}${check};"
-                    continue
+function check_one_server(){
+    local nagios_check_dir=${1}
+    local skip_check_pattern=${2}
+    cd $nagios_check_dir
+    shift
+    shift
+
```
Comment (dennyzhang) [inline bash/enforce_all_nagios_check/enforce_all_nagios_check.sh:17]: Better we assign "$#" to a local variable here, which make the code more clear.

File: bash/enforce_all_nagios_check/enforce_all_nagios_check.sh [hunk 4]
```diff
@@ -1,77 +1,155 @@
...
+    if [ $my_result != "success" ];then
+        nagios_check_result=1
+    fi
+done
+
+echo -ne "==============================================================================\n"
+echo -ne "                               nagios checks end                              \n"
+echo -ne "==============================================================================\n"
+if [ $nagios_check_result == 0 ];then
```
Comment (dennyzhang) [inline bash/enforce_all_nagios_check/enforce_all_nagios_check.sh]: [ $nagios_check_result -eq 0 ]

File: bash/enforce_all_nagios_check/enforce_all_nagios_check.sh (hunk 3 above)
Comment (syrett) [inline bash/enforce_all_nagios_check/enforce_all_nagios_check.sh:17]: I don't understand you, could you please have a example?

nagios_check_dir, skip_check_pattern is already local.

Comment (dennyzhang) [inline bash/enforce_all_nagios_check/enforce_all_nagios_check.sh:17]: Just after two "shift", we add this line:
local my_variable=$#

Then we change L71 to: while [[ $my_variable > 1 ]]

File: bash/enforce_all_nagios_check/enforce_all_nagios_check.sh [hunk 5]
```diff
@@ -1,77 +1,153 @@
...
-fi
-cd $nagios_check_dir
-
-failed_checks=""
-skipped_checks=""
-IFS=$'\n'
-for f in `ls -1 *.cfg`; do
-    if grep '^ *host_name *' $f 2>/dev/null 1>/dev/null; then
-        host_name=$(grep '^ *host_name *' $f | awk -F' ' '{print $2}' | head -n 1)
-        for check in `grep '^ *check_command' $f | awk -F' ' "{print $2}" | awk -F'!' '{print $2}'`; do
```
Comment (syrett) [inline bash/enforce_all_nagios_check/enforce_all_nagios_check.sh]: 注意双引号! 那是个bug

**Pull request merged**